### ISSUE TITLE
deprecated function `hipLaunchKernel`

### ISSUE THREAD

**Issue by psychocoderHPC** [alpaka-group/alpaka#735]
During testing the HIP backend I saw this message.
```
/opt/rocm/include/hip/hcc_detail/functional_grid_launch.hpp:123:3: note: 'hipLaunchKernel<alpaka::vec::Vec<std::integral_constant<unsigned long, 3>, unsigned int>, cupla::CuplaKernel<vectorAdd>, float *, float *, float *, int, void (*)(hip_impl::Empty_launch_parm, alpaka::vec::Vec<std::integral_constant<unsigned long, 3>, unsigned int>, cupla::CuplaKernel<vectorAdd>, float *, float *, float *, int)>' has been explicitly marked deprecated here
[[deprecated("hipLaunchKernel is deprecated and will be removed in the next "
```

Tested with ROCM 2.0.89

**Comment by ax3l** [comment]
cc @tdd11235813

**Comment by psychocoderHPC** [comment]
https://github.com/ROCm-Developer-Tools/HIP/blob/8c5e018e16830432792764489d745dbf33e81cd1/include/hip/hcc_detail/functional_grid_launch.hpp#L151-L153

We need to use `hipLaunchKernelGGL`. 
This function should be available since ROCM 1.6: https://gpuopen.com/whats-new-hip-hcc-rocm-1-6/


btw: the nice thing is that the ROCm docu do not know [hipLaunchKernelGGL](https://rocm-documentation.readthedocs.io/en/latest/search.html?q=hipLaunchKernelGGL&check_keywords=yes&area=default#)  #ILOVEAMD

**Comment by psychocoderHPC** [comment]
documentation: https://github.com/ROCm-Developer-Tools/HIP/blob/e889c04747296dbf8caba13f9ff686a5bec3df45/samples/2_Cookbook/0_MatrixTranspose/Readme.md
